sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -168,6 +168,11 @@ private[parquet] object ParquetFilters {
   def createFilter(schema: StructType, predicate: sources.Filter): Option[FilterPredicate] = {
     val dataTypeOf = getFieldMap(schema)
 
+    // Note that, we do not push down filters for columns having dots in the names. In Parquet
+    // 1.8.2, currently, column APIs in Parquet's `FilterApi` only allow dot-separated names
+    // so here we simply avoid this case. See SPARK-20364.
```

Comment:
Parquet does not allow . in the column name because it is used as a column path delimiter. Since Parquet 1.8.2 (PARQUET-389), Parquet accepts the filter predicates with missing columns. Since it can generate the incorrect results, we block the filter pushdown when the filters have the column having dots in the names. See SPARK-20364.